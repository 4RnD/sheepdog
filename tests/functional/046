#!/bin/bash

# Test vdi deletion

. ./common

for i in 0 1 2; do
    _start_sheep $i
done

_wait_for_sheep 3

_cluster_format -c 2

echo "delete snapshot image before current one"
$COLLIE vdi create test 4G
$COLLIE vdi snapshot test -s snap
_vdi_list

$COLLIE vdi delete test -s snap
$COLLIE vdi delete test -s snap # error
_vdi_list

$COLLIE vdi delete test
$COLLIE vdi delete test # error
_vdi_list

echo "delete current image before snapshot one"
$COLLIE vdi create test 4G
$COLLIE vdi snapshot test -s snap
_vdi_list

$COLLIE vdi delete test
$COLLIE vdi delete test # error
_vdi_list

$COLLIE vdi delete test -s snap
$COLLIE vdi delete test -s snap # error
_vdi_list
