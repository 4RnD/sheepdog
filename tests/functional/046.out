QA output created by 046
using backend plain store
delete snapshot image before current one
  Name        Id    Size    Used  Shared    Creation time   VDI id  Copies  Tag   Block Size Shift
s test         1  4.0 GB  0.0 MB  0.0 MB DATE   7c2b25      6          snap  22
  test         0  4.0 GB  0.0 MB  0.0 MB DATE   7c2b26      6                22
Failed to open VDI test (snapshot id: 0 snapshot tag: snap): Failed to find requested tag
  Name        Id    Size    Used  Shared    Creation time   VDI id  Copies  Tag   Block Size Shift
  test         0  4.0 GB  0.0 MB  0.0 MB DATE   7c2b26      6                22
Failed to open VDI test (snapshot id: 0 snapshot tag: ): No VDI found
  Name        Id    Size    Used  Shared    Creation time   VDI id  Copies  Tag   Block Size Shift
delete current image before snapshot one
  Name        Id    Size    Used  Shared    Creation time   VDI id  Copies  Tag   Block Size Shift
s test         1  4.0 GB  0.0 MB  0.0 MB DATE   7c2b25      6          snap  22
  test         0  4.0 GB  0.0 MB  0.0 MB DATE   7c2b26      6                22
Failed to open VDI test (snapshot id: 0 snapshot tag: ): No VDI found
  Name        Id    Size    Used  Shared    Creation time   VDI id  Copies  Tag   Block Size Shift
s test         1  4.0 GB  0.0 MB  0.0 MB DATE   7c2b25      6          snap  22
Failed to open VDI test (snapshot id: 0 snapshot tag: snap): No VDI found
  Name        Id    Size    Used  Shared    Creation time   VDI id  Copies  Tag   Block Size Shift
